src/04_language.md

```diff
@@ -202,15 +202,161 @@ the process.
+    `a.eq(b)` is equivalent to `a.partial_cmp(b) == Some(std::ordering::Eq)`.
+
+  - *Internal consistency*:
+
+    - `a.lt(b)` iff `a.partial_cmp(b) == Some(std::ordering::Less)`,
+    - `a.gt(b)` iff `a.partial_cmp(b) == Some(std::ordering::Greater)`,
+    - `a.le(b)` iff `a.lt(b) || a.eq(b)`,
+    - `a.ge(b)` iff `a.gt(b) || a.eq(b)`.
+
+    Note that by only defining `partial_cmp`, the internal consistency is
+    guaranteed by the default implementation of `lt`, `le`, `gt`, and `ge`.
+
+  - *Antisymmetry*: `a.lt(b)` (respectively `a.gt(b)`) implies `b.gt(a)`
+    (respectively, `b.lt(b)`). From the developer's standpoint, it also means:
+
+    - `A: PartialOrd<B>`,
+    - `B: PartialOrd<A>`,
+    - both implementations are consistent with each other.
+
+  - *Transitivity*: `a.lt(b)` and `b.lt(c)` implies `a.lt(c)` (also with `gt`,
+    `le` and `ge`). It also means:
+
+    - `A: PartialOrd<B>`,
+    - `B: PartialOrd<C>`,
+    - `A: PartialOrd<C>`,
+    - the implementations are consistent with each other (and their symmetric).
+
+- For `Ord`
+
+  - `PartialOrd<Self>`
+
+  - *Totality*: `a.partial_cmp(b) != None` always. In other words,
+    exactly one of `a.eq(b)`, `a.lt(b)`, `a.gt(b)` is true.
+
+  - *Consistency with `PartialOrd<Self>`*: `Some(a.cmp(b)) == a.partial_cmp(b)`.
+
+The compiler do not check any of those requirements except for the type checking
+itself. However, comparisons are critical because they intervene both in
+liveness critical systems such as schedulers and load balancers, and in
+optimized algorithms that may use `unsafe` blocks.
+In the first use, a bad ordering may lead to availability issues such as
+deadlocks.
+In the second use, it may lead to classical security issues linked to memory
+safety violations. That is again a factor in the practice of limiting the use
+of `unsafe` blocks.
+
+> ### Rule {{#check LANG-CMP-INV | Respect the invariants of standard comparison traits}}
+>
+> In a Rust secure development, the implementation of standard comparison traits
+> must respect the invariants described in the documentation.
+
+<!-- -->
+
+> ### Recommendation {{#check LANG-CMP-DEFAULTS | Use the default method impl. of standard comparison traits}}
+>
+> In a Rust secure development, the implementation of standard comparison traits
+> should only define methods with no default implementation, so as to reduce
+> the risk of violating the invariants associated with the traits.
 
- - The `Send` and `Sync` traits are marker traits that allow implementors
- to specify respectively that a value can be sent (moved) to another thread and
- that a value be shared through a shared reference. They both are unsafe traits,
- that should be used. Moreover, the `unsafe` keyword is necessary for a type
- to implement those traits.
+There is a Clippy lint to check that `PartialEq::ne` is not defined in
+`PartialEq` implementations.
 
- - PartialOrd, Ord, Eq <mark>TODO</mark>
+<mark>TODO</mark> Recommendation: Derive when possible?
```

Comment:
Note that a `#[derive]`d `[Partial]Ord` implementation depends on the order of the fields of the struct (thus making field ordering able to change the semantics of the program)

Reply:
Indeed, it's something that should be addressed in the guide at some point